examples/client-rendering/src/routes/__generated__/Route__Root__Todos__ByStatusDecodedExtra_route.res

```diff
@@ -165,4 +175,28 @@ external makeRenderer: (
   ~prepare: Internal.prepareProps => 'prepared,
   ~prepareCode: Internal.prepareProps => array<RelayRouter.Types.preloadAsset>=?,
   ~render: Internal.renderProps<'prepared> => React.element,
-) => Internal.renderers<'prepared> = ""
\ No newline at end of file
+) => Internal.renderers<'prepared> = ""
+
+
+@live 
+let parseRoute = (route: string, ~exact=false): option<(
+  pathParams,
+  queryParams,
+)> => {
+  switch route->String.split("?") {
+  | [pathName, search] =>
+    RelayRouter.Internal.matchPathWithOptions(
+      {"path": routePattern, "end": exact},
+      pathName,
+    )->Option.map(({params}) => {
+      let params: pathParams = Obj.magic(params)
+      let queryParams =
+        search
+        ->RelayRouter.Bindings.QueryParams.parse
+        ->Internal.parseQueryParams
+      (params, queryParams)
+    })
+  | _ => None
+  }
+}
+
```

Comment:
What's this one used for again?

Regardless, looks like we could put this into an internal utils file and just do a wrapper function calling it here in the generated route to save some bytes. What do you think?

Reply:
Basically I've realized I often want to check if a string matches a given route and if so extract its parameters. Right now this is a pain. This would actually allow to get rid of some other utils functions like `RelayRouter.Utils.isRouteActive`.

I generated it entirely in `_route.res` to avoid some type shenanigans but well we still have an `Obj.magic`, so we could indeed just take care of the types in `_route.res` and have the actual code elsewhere, makes sense.